sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -441,10 +447,10 @@ private static void verifyUnsplittableMethods(ErrorReporter errors, DoFnSignatur
             DoFn.InputProvider.class.getSimpleName(),
             formatType(paramT),
             formatType(expectedInputProviderT));
-        extraParameters.add(DoFnSignature.Parameter.INPUT_PROVIDER);
+        extraParameters.add(DoFnSignature.Parameter.inputProvider());
```

Comment:
Good point, and a good reason to use AutoValue just to be future-proof. Right now it only works because the non-autovalue subclasses are singletons.